### DISCUSSION TITLE
Cannot profile production build

### DISCUSSION THREAD

**Discussion by sarink** [vercel/next.js#13793] [Help]
Since there is currently no way to get source maps on a production build, how are we supposed to profile the application? Profiling in development mode is, obviously, not the same. And if you are in production mode, then you don't get source maps to find where the problem lies.

I know that there is an [experimental](https://github.com/vercel/next.js/pull/13018) feature, but is there anything more reliable, or any workaround for an older version of next? I find it hard to believe that nobody has ever had to do this before 🤔

**Comment by jamesmosier** [comment]
There is a package for that: [next-sourcemaps](https://github.com/vercel/next-plugins/tree/master/packages/next-source-maps)

But that will likely be deprecated after the experimental sourcemaps support is released.

Out of curiosity, why doesn't the experimental feature work for you?

**Comment by sarink** [comment]
Even with next-sourcemaps, it appears functions/classnames are obfuscated. Is there something else that needs to be done? The experimental feature doesn't work because we're on next 9.3.5 (and it only exists in 9.4+)

**Reply by jamesmosier** [reply]
which [`devtool`](https://webpack.js.org/configuration/devtool/) are you using with `next-sourcemaps`?

**Reply by sarink** [reply]
`devtool: 'source-map',`

The source maps are working, that is, I can see a bunch of `.map` files in the output, however, the code is obfuscated, as if it's still being run through uglify (or whatever next uses)? I haven't found a way to disable that (except of course for running in `dev` mode, which kind of defeats the purpose)

**Comment by Vadorequest** [comment]
I have source maps working in production.

You must use `"@zeit/next-source-maps": "0.0.4-canary.1",` specifically, any other version will fail.

https://github.com/UnlyEd/next-right-now/blob/v2-mst-aptd-gcms-lcz-sty/package.json#L144

And, as stated above, this package will be deprecated soon and they'll be integrated within Next.js itself.

**Comment by sarink** [comment]
Yes, I am using the canary version of that library. The source maps are working. The problem seems is that with `dev: false` the code is still obfuscated.

**Reply by Vadorequest** [reply]
I don't use Next.js through a custom server, but using Serverless mode, so I wouldn't know about that, sorry.
